app/services/usps_in_person_proofer.rb

```diff
@@ -0,0 +1,190 @@
+class UspsInPersonProofer
+  attr_reader :token, :token_expires_at
+
+  # Makes a request to retrieve a new OAuth token
+  # and modifies self to store the token and when
+  # it expires (15 minutes).
+  # @return [String] the token
+  def retrieve_token!
+    body = request_token
+    @token_expires_at = Time.zone.now + body['expires_in']
+    @token = "#{body['token_type']} #{body['access_token']}"
+  end
+
+  def token_expired?
+    @token_expires_at.nil? || @token_expires_at.past?
+  end
+
+  # Makes HTTP request to authentication endpoint
+  # and modifies self to store the token and when
+  # it expires (15 minutes).
+  # @return [Hash] API response
+  def request_token
+    url = "#{root_url}/oauth/authenticate"
+    body = {
+      username: AppConfig.env.usps_ipp_username,
+      password: AppConfig.env.usps_ipp_password,
+      grant_type: 'implicit',
+      response_type: 'token',
+      client_id: '424ada78-62ae-4c53-8e3a-0b737708a9db',
+      scope: 'ivs.ippaas.apis',
+    }.to_json
+    resp = faraday.post(url, body, request_headers)
+
+    if resp.success?
+      JSON.parse(resp.body)
+    else
+      { error: 'Failed to get token', response: resp }
+    end
+  end
+
+  # Makes HTTP request to get nearby in-person proofing facilities
+  # Requires address, city, state and zip code.
+  # @param location [Object]
+  # @return [Hash] API response
+  def request_facilities(location)
+    url = "#{root_url}/ivs-ippaas-api/IPPRest/resources/rest/getIppFacilityList"
+    body = {
+      sponsorID: sponsor_id,
+      streetAddress: location.address,
+      city: location.city,
+      state: location.state,
+      zipCode: location.zip_code,
+    }.to_json
+
+    headers = request_headers.merge({
+      'Authorization' => @token,
+      'RequestID' => request_id,
+    })
```

Comment:
style nit, drop the optional curly braces here:

```suggestion
    headers = request_headers.merge(
      'Authorization' => @token,
      'RequestID' => request_id,
    )
```